python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -203,7 +204,11 @@ def get_validated_partition_keys(
         }
         validated_partitions = set()
         for partition_key in partition_keys:
-            multipartition_key = partitions_def.get_partition_key_from_str(partition_key)
+            try:
+                multipartition_key = partitions_def.get_partition_key_from_str(partition_key)
+            except CheckError:
```

Comment:
this except is a little scary... maybe an is_multipartition_key error?

Reply:
yeah i'm happy to change this part